### ISSUE TITLE
map_sides in JSON config file raises exception

### ISSUE THREAD

**Issue by theo-brown** [splewis/get5#642]
### Expected behavior
Loading a match config that contains
```
...
"skip_veto": true, 
"map_sides": ["team2_ct", "team1_ct", "knife"]
...
```
should set the sides for the maps as:
Map 1: Team 1 starting as T, Team 2 starting as CT
Map 2: Team 1 starting as CT, Team 2 starting as T
Map 3: Knife for sides

### Actual behavior
Instead, it raises the following error

```
Exception reported: Invalid Handle 3dd00fe (error 1)
[SM] Blaming: get5.smx
[SM]   [0] StringMap.GetValue
[SM]   [1] Line 69, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json/object.inc::JSON_Object.HasKey
[SM]   [2] Line 82, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json/object.inc::JSON_Object.IsArray.get
[SM]   [3] Line 299, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json.inc::json_cleanup
[SM]   [4] Line 712, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json/object.inc::JSON_Object.Cleanup
[SM]   [5] Line 310, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json.inc::json_cleanup
[SM]   [6] Line 712, /home/splewistx/sm-1.9/addons/sourcemod/scripting/include/json/object.inc::JSON_Object.Cleanup
[SM]   [7] Line 154, ./scripting/get5/matchconfig.sp::LoadMatchFile
[SM]   [8] Line 47, ./scripting/get5/matchconfig.sp::LoadMatchConfig
[SM]   [9] Line 251, ./scripting/get5/matchconfig.sp::SteamWorks_OnMatchConfigReceived
```

### Steps to reproduce

- Plugin version: 0.7.1
- Sourcemod version: 1.10.0.6462
- Steps to reproduce: Create and load a config file containing the `map_sides` key.

**Comment by theo-brown** [comment]
This can also be reproduced with `"map_sides": ["knife", "knife", "knife"]`

**Comment by Drarig29** [comment]
I think some `Cleanup()` calls are missing.

**Comment by splewis** [comment]
Your syntax is correct, that should work, I'll take a look later today and it should be a quick fix.

**Comment by splewis** [comment]
Hm, from a quick test I couldn't reproduce. 

Can you post your full match config file?


> Plugin version: 0.7.1

I'd also suggest trying the latest dev build instead: https://ci.splewis.net/job/get5/lastSuccessfulBuild/ (I should really get around to making a proper release)

**Comment by Drarig29** [comment]
I looked at the JSON Cleanup calls, and they appear to be good. A zone of interest is here:

https://github.com/splewis/get5/blob/master/scripting/get5/matchconfig.sp#L521-L534

But I saw you clean the parent object here:

https://github.com/splewis/get5/blob/master/scripting/get5/matchconfig.sp#L159-L160

And because the `Cleanup()` is recursive, I think everything is cleaned fine. And you don't clean it elsewhere already.

**Comment by Drarig29** [comment]
> Hm, from a quick test I couldn't reproduce.
> 
> Can you post your full match config file?

The discussion started here, you can maybe try with the config file in here: https://github.com/splewis/get5/discussions/641

**Comment by theo-brown** [comment]
Yep, this is [the config](https://github.com/splewis/get5/discussions/641) I've been using. I've altered it slightly:
```
{"matchid": "Team A vs Team B",
 "num_maps": 3, 
"maplist": ["de_mirage", "de_nuke", "de_vertigo"], 
"skip_veto": true, 
"map_sides": ["team1_ct", "team2_ct", "knife"],
"team1": {"name": "Team A", "players": ["76561198922288040"]}, 
"team2": {"name": "Team B", "players": ["76561198365483001"]}}

**Comment by theo-brown** [comment]
Ok, I've updated the plugin, and it seems to be working.
**Conclusion**
`match_sides` works in get5 version 0.7.2-dev+520
`match_sides` does not work in get5 version 0.7.1

**Comment by theo-brown** [comment]
Might be worth:
- Adding a note in docs somewhere that notes that it doesn't work in 0.7.1 (if anyone's been able to / has bothered to replicate this in 0.7.1)
- Packaging a new release

**Comment by TandelK** [comment]
@theo-brown its a duplicate issue which i reported way back. https://github.com/splewis/get5/issues/447 

It has been solved with new Master branch but not yet released in release versions.

**Comment by theo-brown** [comment]
> I should really get around to making a proper release

@splewis is that something that I or another contributor could do?
Thanks!

**Comment by theo-brown** [comment]
@splewis is this fixed in release 0.7.2?

**Comment by TandelK** [comment]
@theo-brown Yes of course.
